pkg/controller/vm_webhook.go

```diff
@@ -296,9 +330,23 @@ func ValidateMemory(ctx context.Context, memory *virtv1alpha1.Memory, fieldPath
 		errs = append(errs, field.Required(fieldPath.Child("size"), ""))
 		return errs
 	}
-	if memory.Size.Value() <= 0 {
-		errs = append(errs, field.Invalid(fieldPath.Child("size"), memory.Size.Value(), "must be greater than 0"))
+	memSize := memory.Size.Value()
+	if memSize <= 0 {
+		errs = append(errs, field.Invalid(fieldPath.Child("size"), memSize, "must be greater than 0"))
 	}
+	if memory.Hugepages != nil {
+		switch memory.Hugepages.PageSize {
```

Comment:
We could parse pagesize to integer, and try mod memsize with it? So that we can remove this switch-case clause?